tests/integration/test_integration.py

```diff
@@ -418,3 +418,37 @@ def test_flush_log(caplog):
             )
         ]
         log.log.assert_has_calls(calls)
+
+
+def test_regression_logging_in_context(tmpdir):
+    """
+    When logs injection is enabled and the logger is patched
+        When a parent span closes before a child
+            The application does not deadlock due to context lock acquisition
+    """
+    f = tmpdir.join("test.py")
+    f.write(
+        """
+import ddtrace
+ddtrace.patch(logging=True)
+
+s1 = ddtrace.tracer.trace("1")
+s2 = ddtrace.tracer.trace("2")
+s1.finish()
+s2.finish()
+""".lstrip()
+    )
+    p = subprocess.Popen(
+        [sys.executable, "test.py"],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        cwd=str(tmpdir),
+        env=dict(
+            DD_TRACE_LOGS_INJECTION="true",
```

Comment:
could be worth having this be a parametrized test, we should expect this test case to pass regardless of these env variables

maybe the same with `ddtrace.patch(logging=True)` ?